### PR TITLE
Demo of generating swagger file from code and added response header info

### PR DESCRIPTION
**Pull request by geoffsmith82**
Hi Marcelo,

Here is my next contribution. I've added response header handling and generating a swagger json file from code. The area that could do with some improvement is schema/$ref handling. My demo shows how it currently works though.

### CODE REVIEW

**Comment by marcelojaloto** [comment]
@geoffsmith82 ,
i found schema error in the "lincense" additional property.

image

Take your swagger json demo file and put in swagger editor for look the error.

https://editor.swagger.io

Do you know swagger spec version 2,0? 

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md

image

The "lincense" additional property belongs to info object. 

Look at the example solved in the swagger editor.
image

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 1]
```diff
@@ -0,0 +1,191 @@
+unit frmGenerate;
+
+interface
+
+uses
+  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: each unit in a row.

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  end;
+
+var
+  Form1: TForm1;
+
+implementation
+
+{$R *.dfm}
+
+uses REST.Json,
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: each unit in a row.

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  Swag.Doc.Path,
+  Swag.Doc.Path.Operation,
+  Swag.Doc.Path.Operation.RequestParameter,
+  Swag.Doc.Path.Operation.Response,
+  Swag.Doc.Definition,
+  Swag.Doc;
+
+procedure TForm1.btnGenerateClick(Sender: TObject);
+var
+  vSwagDoc : TSwagDoc;
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: code pattern:
```
var
  vVariable: TType;
```

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    vSwagDoc.Produces.Add('text/xml');
+    vSwagDoc.Produces.Add('application/json');
+
+    vSwagDoc.Schemes := [tpsHttps];
+
+    vDefinition3 := TSwagDefinition.Create;
+    vDefinition3.Name := 'SomeSubType';
+
+    vJSONSchema := TJSONObject.Create;
+    vJSONSchema.AddPair('type','object');
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: code pattern.

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    vJSONProperities.AddPair('id',vJSONType);
+
+    vJSONSchema.AddPair('properties',vJSONProperities);
+
+
+    vDefinition3.JsonSchema := vJSONSchema;
+    
+    vSwagDoc.Definitions.Add(vDefinition3);
+
+
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: code pattern: no excess space

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    vParam.Required := False;
+    vParam.TypeParameter := 'string';
+    vOperation.Parameters.Add(vParam);
+
+    vParam := TSwagRequestParameter.Create;
+    vParam.Name := 'param3';
+    vParam.InLocation := rpiBody;
+    vParam.Required := True; 
+
+    vDefinition2 := TSwagDefinition.Create;
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: memory leak

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+    vJSONType := TJSONObject.Create;
+    vJSONType.AddPair('type','integer');
+    vJSONType.AddPair('format','int64');
+
+    vJSONProperities := TJSONObject.Create;
+    vJSONProperities.AddPair('id',vJSONType);
+    
+
+    vJSONType := TJSONObject.Create;
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: memory leak

File: Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+    vOperation.Parameters.Add(vParam);
+
+    vParam := TSwagRequestParameter.Create;
+    vParam.Name := 'param3';
+    vParam.InLocation := rpiBody;
+    vParam.Required := True; 
+
+    vDefinition2 := TSwagDefinition.Create; 
+    vDefinition2.Name := 'SomeType';
+    vParam.Schema.JsonSchema := vDefinition2.NameToJson;
```
Comment (marcelojaloto) [inline Demos/GenerateSwaggerJsonFromCode/frmGenerate.pas]: Use property "Name" of the schema object. SwagDoc library automatically creates a ref definition schema json.

File: Demos/LoadSwaggerJsonToObject/frmSimpleDemo.pas [hunk 9]
```diff
@@ -28,11 +28,14 @@ implementation
 
 {$R *.dfm}
 
-uses REST.Json;
+uses REST.Json, System.IOUtils;
```
Comment (marcelojaloto) [inline Demos/LoadSwaggerJsonToObject/frmSimpleDemo.pas]: code pattern

File: Demos/LoadSwaggerJsonToObject/frmSimpleDemo.pas [hunk 10]
```diff
@@ -28,11 +28,14 @@ implementation
...
 
 procedure TfrmSimpleSwaggerDocDemo.btnLoadJSONClick(Sender: TObject);
+var
+  vFullFilename : String;
```
Comment (marcelojaloto) [inline Demos/LoadSwaggerJsonToObject/frmSimpleDemo.pas]: code pattern

**Comment by geoffsmith82** [comment]
Hi Marcelo,
I have fixed up my code and it now validates properly in the swagger editor.  I have been using the 
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md , but obviously put that in the wrong place.

**Comment by marcelojaloto** [comment]
@geoffsmith82 (Geoffrey Smith),
thank you a lot for more contribuitions at the SwagDoc.

Your contribuitions are very important for the SwagDoc.

Congratulation!

**Review by marcelojaloto — COMMENTED** [review]
fix code pattern and memory leak.

**Comment by marcelojaloto** [comment]
@cunhaw 
can you to review?

**Review by cunhaw — APPROVED** [review]
Very good!

**Pull request merged**